### ISSUE TITLE
Terminal can essentially become bricked after pressing ctrl+s

### ISSUE THREAD

**Issue by Tyriar** [microsoft/vscode#168192]
On Linux/macOS if you press ctrl+s it will pause the pty, the way to undo this is to press ctrl+q but we route that to a vscode keybinding so it essentially bricks the terminal. See https://github.com/microsoft/terminal/issues/13217

**Comment by meganrogge** [comment]
That doesn't happen for me on macOS 🤔 I can press escape or ctrl and get out of it

**Comment by Tyriar** [comment]
I reproduced it on WSL at least. Oh on macOS we wouldn't bind ctrl+q so it would work there, though there's a discoverability issue because ctrl+s might accidentally get hit.

**Comment by meganrogge** [comment]
Ok just fyi, doesn't repro on WSL for me either

**Comment by Tyriar** [comment]
@meganrogge does ctrl+s stop all input? If so, doesn't ctrl+q open this for you?

![image](https://user-images.githubusercontent.com/2193314/205996275-0212bde7-c09f-41ae-a566-1ade016d3b0f.png)

**Comment by meganrogge** [comment]
in both cases with the terminal focused:

on macOS, `ctrl+s` triggers `fwd-i-search` which can be escaped

on wsl, `ctrl+s` does nothing

**Comment by zadjii-msft** [comment]
I got chu

![vscode-168192](https://user-images.githubusercontent.com/18356694/205999694-e1bf42f9-1fd2-40dc-9b28-0524b20f86e8.gif)

**Comment by Tyriar** [comment]
Yeah that's what I see. My proposal is we add a warning status, maybe with some input-related icon (`keyboard`?) and an action that appears in the hover to send ctrl+q

**Comment by Tyriar** [comment]
Opening up for contributions, I was thinking we could have a simply [`terminalContrib/`](https://github.com/microsoft/vscode/blob/faacc5f4805c24f23258b6b305c0e60089b78330/src/vs/workbench/contrib/terminalContrib) which only listens to whether the ctrl+s mode is on and shows a status bar item while the terminal is focused. I'm not sure whether determining if the mode is on needs new API exposed in [xterm.js](https://github.com/xtermjs/xterm.js)

**Comment by PKNaveen** [comment]
I tried replicating this on the version 1.90.2, but I couldn't. Has this been resolved?

Version: 1.90.2 (user setup)
Commit: 5437499feb04f7a586f677b155b039bc2b3669eb
Date: 2024-06-18T22:34:26.404Z
Electron: 29.4.0
ElectronBuildId: 9728852
Chromium: 122.0.6261.156
Node.js: 20.9.0
V8: 12.2.281.27-electron.0
OS: Windows_NT x64 10.0.22631

**Comment by nekodojo** [comment]
I had this problem too, but I notice it is only for shell windows that I open in the editor (top) space.  Just pressing Ctrl-backtick gives me a terminal in the bottom panel that does NOT freeze on Ctrl-S
